ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+Playlist.swift

```diff
@@ -363,6 +363,45 @@ extension BrowserViewController: PlaylistScriptHandlerDelegate,
   ) {
     PlaylistP3A.recordUsage()
 
+    let addItemToPlaylist = {
+      [weak self] (
+        item: PlaylistInfo,
+        folderUUID: String?,
+        completion: ((_ didAddItem: Bool) -> Void)?
+      ) in
+      PlaylistItem.addItem(item, folderUUID: folderUUID, cachedData: nil) {
+        guard let self = self else { return }
+
+        if let url = URL(string: item.src), url.scheme == "blob" {
+          // Spawn a WebView to load the non-blob asset
+          Task { @MainActor in
+            let mediaStreamer = PlaylistMediaStreamer(
+              playerView: self.view,
+              webLoaderFactory: LivePlaylistWebLoaderFactory()
+            )
+
+            let newItem = (try? await mediaStreamer.loadMediaStreamingAsset(item)) ?? item
+            PlaylistManager.shared.autoDownload(item: newItem)
```

Comment:
We should just not bother auto-downloading if `loadMediaStreamingAsset` fails since its not going to work anyways

Reply:
Fixed.